### ISSUE TITLE
Connecting to VLLM OpenAI API Compatible Server

### ISSUE THREAD

**Issue by JoshuaFurman** [weaviate/Verba#123]
In my lab environment i am serving mixtral with VLLM using their OpenAI API compatible server and I'm hosting a weaviate instance as well. 

I just spun up verba, pointing to both my weaviate instance and VLLM instance using the .env file and connection to weaviate seems to be all good. I can see my schema and object count in the status tab but any queries i make seem to break... Unsure if this is a limitation on being able to handle models other than GPT-3.5 || GPT-4 being served from OpenAI. 

Has anyone been able to configure a setup like this?

Thanks!

**Comment by JoshuaFurman** [comment]
Looks as though you are boxed in to use either ADA embedding from OpenAI or MiniLM or Cohere as well...

**Comment by thomashacker** [comment]
Yes, right now there is no support for Mixtral models! But great point, we'll look into that for the next update

**Comment by samos123** [comment]
I got this working end to end but had to make some changes to be able to use my custom embedding model server. I submitted a PR for the changes I needed to be able to use OpenAI compatible API server for both embeddings and LLM: https://github.com/weaviate/Verba/pull/148

I plan to publish an end to end tutorial that runs on K8s to install Verba, Weaviate, an LLM and an embedding model server all within the same cluster. Stay tuned!

**Comment by samos123** [comment]
I finished writing my guide for end-to-end private Verba RAG using Weaviate, Lingo, vLLM + Mistral 7b v2 and Sentence Transformers: https://www.substratus.ai/blog/lingo-weaviate-private-rag

Looking forward to hearing feedback. The guide should help you with figuring out how to use vanilla vLLM with Verba too.

**Comment by gcheron** [comment]
I am using the VLLM OpenAI API Compatible Server to serve Llama3 with Verba.

@thomashacker, this issue is set as **completed**, but are we supposed to be able to set models other than OpenAI ones (GPT-4, GPT-3.5)?
When I set OpenAI URL to point to my instance (e.g., `OPENAI_BASE_URL=http://localhost:7000/v1`), I can only see OpenAI models while my server is serving Llama3:
![image](https://github.com/user-attachments/assets/a274e079-8621-4101-8da5-af5217195ec6)

@samos123 in your tutorial, you specified the following content in a file named `verba-values.yaml`:

```
env:
- name: OPENAI_MODEL
  value: mistral-7b-instruct-v0.2
```

Where does the `OPENAI_MODEL` variable come from?

I couldn't find this variable in the `Verba` codebase, which makes me think that even when using the VLLM OpenAI API Compatible Server with Verba, it only supports OpenAI models (e.g., GPT-4, GPT-3.5).

**Comment by samos123** [comment]
Looks like the codebase has indeed changed. @thomashacker may know the right approach to do this with latest codebase.

**Comment by thomashacker** [comment]
Good point! For the `OpenAIGenerator` it's was actually still hardcoded to default models. I just merged this PR (https://github.com/weaviate/Verba/pull/363) that allows Verba to dynamically retrieve model names from custom OpenAI Server set in the `OPENAI_BASE_URL` in your `.env`. 

Let me know if this works for your usecase! 🚀

**Comment by gcheron** [comment]
Thank you for your prompt response!
I’ve implemented custom changes that are currently working well for my use case.
I’ll review your new implementation soon and compare it with mine to see if it aligns with my requirements (I think I need some extra customizability).
If not, I can submit a PR with my adjustments.

**Comment by thomashacker** [comment]
That'd be amazing! Thanks a lot

**Comment by gcheron** [comment]
@thomashacker sorry for the delay but here is the PR https://github.com/weaviate/Verba/pull/373
